Breeding-Insight/bi-api
# Bug/bi 1852

# Description
**Story:**[ BI-1852](https://breedinginsight.atlassian.net/browse/BI-1852) - Duplicate breeding method can be created

Definition: A Breeding Method is a 'duplicate' if: there already exist a breeding method with the same _name_ or _abbreviation_


# Dependencies
bi-web: bug/BI-1852

# Testing
**Set up**
1. Select a program for which the current user is a _breeder_
2. Go to the Program Administration page
3. Select the _Breeding Methods_ tab

**TEST 1 (create new breeding method)**
1. Select the _Create Breeding Method_ button
2. Create a new breeding method with the same name or abbreviation as an existing breeding method.

EXPECTED RESULT
- An Error banner with the text:
-- `A breeding method with the name '<name>' is already defined in the system.`
or
-- `A breeding method with the abbreviation '<abbreviation>' is already defined in the system.`
- No new breeding method was created.
3. Select the _Create Breeding Method_ button
4. Create a new breeding method with a unique name and abbreviation.

EXPECTED RESULT
- A Success banner is displayed
- New breeding method was created and displayed.


**TEST 2 (edit existing breeding method)**
1. Edit an existing _program_ breeding method.
2. Modify the _name_ or _abbreviation_ fields to duplicate an existing _name_ or _abbreviation_ .
3. Select the _Save_ button

EXPECTED RESULT
- An Error banner with the text:
-- `A breeding method with the name '<name>' is already defined in the system.`
or
-- `A breeding method with the abbreviation '<abbreviation>' is already defined in the system.`
- Changes were not saved.
5. Edit an existing _program_ breeding method.
6. Modify the _name_ or _abbreviation_ fields be new unique (non-duplicate) values.
7. Select the _Save_ button

EXPECTED RESULT
- A Success banner is displayed
- Modified data is listed.

# Checklist:

- [x] I have run TAF: _\<[please include a link to TAF run](https://github.com/Breeding-Insight/taf/actions/runs/6316931011)>_
